apps/tunes/management/commands/tunes_create_songs_from_spotify.py

```diff
@@ -14,76 +15,96 @@ def __init__(self, *args, **kwargs):
         """
         success, fail = 0, 0
         for track in tracks:
 
-            song, created = Song.objects.get_or_create(code=track['code'], defaults=track)
-
-            if created:
-                msg = 'Created song with code {}'.format(song.code)
-                self.write_to_log_and_output(msg)
-                success += 1
-            else:
-                msg = 'Song with code {} already exists'.format(song.code)
-                self.write_to_log_and_output(msg, output_stream='stderr')
+            try:
+                song, created = Song.objects.get_or_create(code=track['code'], defaults=track)
+
+                if created:
+                    msg = 'Created song with code {}'.format(song.code)
+                    self.write_to_log_and_output(msg)
+                    success += 1
+                else:
+                    msg = 'Song with code {} already exists'.format(song.code)
+                    self.write_to_log_and_output(msg)
+                    fail += 1
+            except ValidationError:
+                msg = 'ERROR: Could not create song with data: {}'.format(track)
+                self.write_to_log_and_output(msg, output_stream='stderr', log_level='warning')
                 fail += 1
 
         return success, fail
 
-    def handle(self, *args, **options):
-        self.logger.info('{} - Starting run to create songs from Spotify'.format(self._unique_id))
-
-        total_songs = options.get('total_songs')
-        num_playlists = 10
-        max_tracks_from_playlist = settings.SPOTIFY['max_songs_from_list']
-        max_tracks_from_category = settings.SPOTIFY['max_songs_from_category']
+    def get_tracks_from_spotify(self):
+        """
+        Request, format, and return tracks from Spotify's API.
 
+        :return: (list[dict]) Track data for saving as Song records
+        """
         spotify = SpotifyClient(command_id=self._unique_id)
-
         tracks = []
 
         for category in settings.SPOTIFY['categories']:
             songs_from_category = 0
 
-            playlists = spotify.get_playlists_for_category(category, num_playlists)
+            try:
+                playlists = spotify.get_playlists_for_category(category, settings.SPOTIFY['max_playlist_from_category'])
+                self.write_to_log_and_output('Got {} playlists for category: {}'.format(len(playlists), category))
 
-            for playlist in playlists:
-                raw_tracks = spotify.get_songs_from_playlist(playlist, max_tracks_from_playlist)
-                new_tracks = spotify.get_audio_features_for_tracks(raw_tracks)
+                for playlist in playlists:
+                    if songs_from_category < settings.SPOTIFY['max_songs_from_category']:
+                        num_tracks = settings.SPOTIFY['max_songs_from_category'] - songs_from_category
+                        raw_tracks = spotify.get_songs_from_playlist(playlist, num_tracks)
```

Comment:
Add logging around making calls to Spotify to ensure we have an audit trail of what calls have been made and what call resulted in an error